src/context.hpp

```diff
@@ -41,6 +43,17 @@ struct cvk_context : public _cl_context,
             }
             m_properties.push_back(*props);
         }
+        // Get buffer size from extension.
+        auto buff_size_prop_index =
+            get_property_index(CL_PRINTF_BUFFERSIZE_ARM);
+        if (buff_size_prop_index != -1) {
+            m_buffer_size = m_properties[buff_size_prop_index];
+        }
+        // If buffersize is set in config then default to that.
+        if constexpr (std::is_member_object_pointer_v<
+                          decltype(&config_struct::printf_buffer_size)>) {
+            m_buffer_size = config.printf_buffer_size;
+        }
```

Comment:
When I was saying printf buffer size is set, I was referering to https://github.com/kpet/clvk/blob/main/src/config.hpp#L42

No need for `type_traits`. Just check for `config.printf_buffer_size.set`. In fact `config.printf_buffer_size` is a shortcut for `config.printf_buffer_size()` or `config.printf_buffer_size.value`.

And make sure to always set `m_buffer_size`. With the current implementation, it's not obvious that we would not end up with an uninitialized value.